beacon_node/execution_layer/src/lib.rs

```diff
@@ -606,132 +652,193 @@ impl<T: EthSpec> ExecutionLayer<T> {
+                            Ok(ProvenancedPayload::Local(local))
                         }
                         (Ok(Some(relay)), Ok(local)) => {
-                            let is_signature_valid = relay.data.verify_signature(spec);
-                            let header = relay.data.message.header;
+                            let header = &relay.data.message.header;
 
                             info!(
                                 self.log(),
-                                "Received a payload header from the connected builder";
-                                "block_hash" => ?header.block_hash(),
+                                "Received local and builder payloads";
+                                "relay_block_hash" => %header.block_hash(),
+                                "local_block_hash" => %local.block_hash(),
+                                "parent_hash" => %parent_hash,
                             );
 
-                            let relay_value = relay.data.message.value;
-                            let configured_value = self.inner.builder_profit_threshold;
-                            if relay_value < configured_value {
-                                info!(
+                            match verify_builder_bid(
+                                &relay,
+                                parent_hash,
+                                prev_randao,
+                                timestamp,
+                                Some(local.block_number()),
+                                self.inner.builder_profit_threshold,
+                                spec,
+                            ) {
+                                Ok(()) => {
+                                    Ok(ProvenancedPayload::Builder(relay.data.message.header))
+                                }
+                                Err(reason) => {
+                                    metrics::inc_counter_vec(
+                                        &metrics::EXECUTION_LAYER_GET_PAYLOAD_BUILDER_REJECTIONS,
+                                        &[reason.as_ref().as_ref()],
+                                    );
+                                    warn!(
                                         self.log(),
-                                        "The value offered by the connected builder does not meet \
-                                        the configured profit threshold. Using local payload.";
-                                        "configured_value" => ?configured_value, "relay_value" => ?relay_value
+                                        "Builder returned invalid payload";
+                                        "info" => "using local payload",
+                                        "reason" => %reason,
+                                        "relay_block_hash" => %header.block_hash(),
+                                        "parent_hash" => %parent_hash,
                                     );
```

Comment:
I think maybe this should only be an `info` log in the case the bid threshold isn't met. Otherwise we'll get a lot of noisy warnings (better than noisy crits, but still).

Reply:
Good call, I've made it so it's only an info when the payload is valid but ignored: e0b2b1aa4

I've maintained the `warn` for invalid payloads since I'm keen to be loud about builders that are doing the wrong thing.